distros/ros2-overlay.nix

```diff
@@ -172,9 +172,11 @@ rosSelf: rosSuper: with rosSelf.lib; {
       # Use X11 by default in RViz2.
       # https://github.com/ros-visualization/rviz/issues/1442
       "--set-default QT_QPA_PLATFORM xcb"
+      "--set QT_QPA_PLATFORM xcb"
```

Comment:
I use rviz2 under wayland without problems. I guess your problem is that you have set `QT_QPA_PLATFORM` in your environment to something else than `xcb`. In this case, the `--set-default` line has no effect. If you unset `QT_QPA_PLATFORM`, it should work.

I'm not sure whether hardcoding `QT_QPA_PLATFORM` to `xcb` is a good thing, because if prevents users from overriding that. I'm not sure whether it makes sense to use `rviz2` with other QPA platforms. If not, that we can hardcode xcb, but maybe people on MacOS would need to use something else.

Anyway, if we hardcode `xcb` here, I'd just replace `--set-default` with `--set` instead of adding the new line.